VMEncryption/main/ResourceDiskUtil.py

```diff
@@ -0,0 +1,274 @@
+        if not self.resource_disk_partition_exists():
+            self.logger.log('LUKS format operation requested, but resource disk partition does not exist')
+            return False
+        return (int)(self.disk_util.luks_format(passphrase_file=self.RD_KEY_FILE, dev_path=self.RD_DEV_PATH, header_file=None)) == 0
+
+    def encrypt(self):
+        """ use disk util with the appropriate device mapper """
+        self.mount_key_volume()
+        return (int)(self.disk_util.encrypt_disk(dev_path=self.RD_DEV_PATH, passphrase_file=self.RD_KEY_FILE, mapper_name=self.mapper_name, header_file=None)) == 0
+
+    def make(self):
+        """ make a default file system on top of the crypt layer """
+        make_result = self.disk_util.format_disk(dev_path=self.mapper_path, file_system=CommonVariables.default_file_system)
+        if make_result != CommonVariables.process_success:
+            self.logger.log(msg="Failed to make file system on ephemeral disk", level=CommonVariables.ErrorLevel)
+            return False
+        # todo - drop DATALOSS_WARNING_README.txt file to disk
+        return True
+
+    def mount_key_volume(self):
+        """ attempt to mount the key volume and verify existence of key file"""
+        if not os.path.exists(self.RD_KEY_FILE):
+            self.disk_util.make_sure_path_exists(self.RD_KEY_FILE_MOUNT_POINT)
+            key_volume_device_name = os.popen('blkid -L "' + self.RD_KEY_VOLUME_LABEL + '"').read().strip()
+            self.disk_util.mount_filesystem(key_volume_device_name, self.RD_KEY_FILE_MOUNT_POINT)
+        return os.path.exists(self.RD_KEY_FILE)
+        
+    def mount(self):
+        """ mount the file system previously made on top of the crypt layer """
+        #ensure that resource disk mount point directory has been created
+        cmd = 'mkdir -p ' + self.RD_MOUNT_POINT
+        if self.executor.Execute(cmd, suppress_logging=True) != CommonVariables.process_success:
+            self.logger.log(msg='Failed to precreate mount point directory: ' + cmd, level=CommonVariables.ErrorLevel)
+            return False
+
+        # mount to mount point directory
+        mount_result = self.disk_util.mount_filesystem(dev_path=self.mapper_path, mount_point=self.RD_MOUNT_POINT, file_system=CommonVariables.default_file_system)
+        if mount_result != CommonVariables.process_success:
+            self.logger.log(msg="Failed to mount file system on resource disk", level=CommonVariables.ErrorLevel)
+            return False
+        return True
+
+    def configure_waagent(self):
+        """ turn off waagent.conf resource disk management  """ 
+        # set ResourceDisk.MountPoint to standard /mnt mount point
+        cmd = "sed -i.bak 's|ResourceDisk.MountPoint=.*|ResourceDisk.MountPoint=/mnt|' /etc/waagent.conf"
+
+        # set ResourceDiskFormat=n to ensure waagent does not attempt a simultaneous format
+        cmd = "sed -i.bak 's|ResourceDisk.Format=y|ResourceDisk.Format=n|' /etc/waagent.conf"
```

Comment:
There do not seem to be any existing .bak backups of the waagent.conf file, however in other areas where this is done, a unique identifier is added to the extension so as to avoid conflicts or overwrites.  I will look into carrying over some of that logic here as well.